### PR TITLE
logger: remove valist

### PR DESCRIPTION
**Pull request by hawkingrei**
remove Valist and make logger return the log string instead of ```snprinf``` by client.

Signed-off-by: Wangweizhen <hawking.rei@gmail.com>

### CODE REVIEW

**Comment by hawkingrei** [comment]
All green. add test for the length of log string.

File: src/logger.rs [hunk 1]
```diff
@@ -16,11 +16,11 @@ extern "C" fn destructor(ctx: *mut c_void) {
...
 }
 
-extern "C" fn logv(ctx: *mut c_void, log_level: InfoLogLevel, format: *const c_char, ap: VaList) {
+extern "C" fn logv(ctx: *mut c_void, log_level: InfoLogLevel, log: *const c_char) {
     unsafe {
         let logger = &*(ctx as *mut Box<dyn Logger>);
-        let format = CStr::from_ptr(format);
-        logger.logv(log_level, format.to_str().unwrap(), ap);
+        let log = CStr::from_ptr(log);
+        logger.logv(log_level, log.to_str().unwrap());
```
Comment (yiwu-arbug) [inline src/logger.rs:23]: Let's do `log.to_string_lossy()` here to avoid unexpected panic.

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -226,16 +226,22 @@ struct crocksdb_logger_impl_t : public Logger {
   void* rep;
 
   void (*destructor_)(void*);
-  void (*logv_internal_)(void* logger, int log_level, const char* format,
-                         va_list ap);
+  void (*logv_internal_)(void* logger, int log_level, const char* log);
+
+  void log_help_(void* logger, int log_level, const char* format, va_list ap) {
```
Comment (yiwu-arbug) [inline librocksdb_sys/crocksdb/c.cc:231]: struct can have private method:
```
 private:
  void log_help(...){...}
```

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -226,16 +226,22 @@ struct crocksdb_logger_impl_t : public Logger {
...
+    const int kBufferSize = 1024;
```
Comment (yiwu-arbug) [inline librocksdb_sys/crocksdb/c.cc:232]: nit: s/const/constexpr/

Comment (hawkingrei) [reply librocksdb_sys/crocksdb/c.cc]: Done

File: src/logger.rs (hunk 1 above)
Comment (hawkingrei) [reply src/logger.rs]: Done

**Comment by hawkingrei** [comment]
@yiwu-arbug PTAL

**Pull request merged**